vars/cloudFoundryDeploy.groovy

```diff
@@ -222,3 +234,33 @@ Transformed manifest file content: $transformedManifest"""
         writeYaml file: config.cloudFoundry.manifest, data: manifest
     }
 }
+
+
+private void reportToInflux(script, config, deploySuccess, JenkinsUtils jenkinsUtils) {
+    def deployUser = ''
+    withCredentials([usernamePassword(
+        credentialsId: config.cloudFoundry.credentialsId,
+        passwordVariable: 'password',
+        usernameVariable: 'username'
+    )]) {
+        deployUser = username
+    }
+
+    def timeFinished = new Date().format( 'MMM dd, yyyy - HH:mm:ss' )
+    def triggerCause = jenkinsUtils.isJobStartedByUser()?'USER':(jenkinsUtils.isJobStartedByTimer()?'TIMER': 'OTHER')
+
+    def deploymentData = [deployment_data: [
+        artifactUrl: 'n/a', //might be added later on during pipeline run (written to commonPipelineEnvironment)
+        deployTime: timeFinished,
+        jobTrigger: triggerCause
+    ]]
+    def deploymentDataTags = [deployment_data: [
+        artifactVersion: script.commonPipelineEnvironment.getArtifactVersion(),
+        deployUser: deployUser,
+        deployResult: deploySuccess?'SUCCESS':'FAILURE',
+        cfApiEndpoint: config.cloudFoundry.apiEndpoint,
+        cfOrg: config.cloudFoundry.org,
+        cfSpace: config.cloudFoundry.space,
+    ]]
+    writeInflux script: script, customData: [:], customDataTags: [:], customDataMap: deploymentData, customDataMapTags: deploymentDataTags
```

Comment:
Does it introduce a new plugin dependency? If yes, then it would break the deployment step.